### PR TITLE
receivers: populate author_suggest

### PR DESCRIPTION
**Pull request by iulianav**
Signed-off-by: Iuliana Voinea <iulianavoinea96@gmail.com>

## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/records/receivers.py [hunk 1]
```diff
@@ -178,12 +178,35 @@ def enhance_after_index(sender, json, *args, **kwargs):
...
+    """Populate the ``author_suggest`` field of Authors records."""
+    if 'authors.json' not in json.get('$schema'):
+        return
+
+    author_paths = [
+        'name.preferred_name',
+        'name.value',
+        'native_name',
+        'other_names',
+        'previous_names',
```
Comment (michamos) [inline inspirehep/modules/records/receivers.py]: seems OK, note that this is going to change with https://github.com/inspirehep/inspire-schemas/pull/182, where all those names will be under `name`.

File: inspirehep/modules/records/receivers.py [hunk 2]
```diff
@@ -178,12 +178,35 @@ def enhance_after_index(sender, json, *args, **kwargs):
...
     populate_name_variations(sender, json, *args, **kwargs)
     populate_title_suggest(sender, json, *args, **kwargs)
 
 
+def populate_author_suggest(sender, json, *args, **kwargs):
+    """Populate the ``author_suggest`` field of Authors records."""
+    if 'authors.json' not in json.get('$schema'):
+        return
+
+    author_paths = [
```
Comment (iulianav) [inline inspirehep/modules/records/receivers.py]: @michamos we will leave it like this, ready for the new schema and merge it when the schema is stable and through. Looks ok?

File: inspirehep/config.py [hunk 3]
```diff
@@ -338,6 +338,13 @@
         'application/json': 'invenio_records_rest.serializers:json_v1_search',
         'application/vnd+inspire.ids+json': 'inspirehep.modules.api.v1.common_serializers:json_recids_response',
     },
+    'suggesters': {
+        'author': {
+            'completion': {
```
Comment (iulianav) [inline inspirehep/config.py]: ` '_source': False ` will have to be added before the `completion` key in order to support source filtering IF https://github.com/inveniosoftware/invenio-records-rest/pull/226 gets merged.

**Comment by drjova** [comment]
Supersede by https://github.com/inspirehep/inspire-next/pull/3650

**Pull request closed without merging**